### PR TITLE
formItem support error message with reactNode

### PR DESCRIPTION
**Pull request by zombieJ**
support code like:
```jsx
form.setFields({
  title: {
    errors: [
      <div>My Error</div>,
    ]
  }
});
```

ref: #13014

### CODE REVIEW

**Comment by afc163** [comment]
Rebase

**Comment by zombieJ** [comment]
rebased. Seems ts-lint updated?

**Comment by afc163** [comment]
https://github.com/ant-design/antd-tools/pull/80

https://github.com/ant-design/ant-design/pull/13060

File: components/form/FormItem.tsx [hunk 1]
```diff
@@ -72,11 +72,17 @@ export default class FormItem extends React.Component<FormItemProps, any> {
     if (help === undefined && this.getOnlyControl()) {
       const errors = this.getField().errors;
       if (errors) {
-        return intersperse(errors.map((e: any, index: number) => (
-          React.isValidElement(e.message)
-            ? React.cloneElement(e.message, { key: index })
-            : e.message
-        )), ' ');
+        return intersperse(errors.map((e: any, index: number) => {
```
Comment (ztplz) [inline components/form/FormItem.tsx]: Consider dropping `intersperse` since it's useless.

File: components/form/FormItem.tsx [hunk 2]
```diff
@@ -26,6 +25,14 @@ export interface FormItemProps {
   colon?: boolean;
 }
 
+function intersperseSpace<T>(list?: Array<T>): Array<T | string> {
```
Comment (ztplz) [inline components/form/FormItem.tsx]: `list` isn't a optional parameter, remove `?`.

**Review by ztplz — APPROVED** [review]

File: components/form/FormItem.tsx [hunk 3]
```diff
@@ -26,6 +25,14 @@ export interface FormItemProps {
...
+  return (list || [])
```
Comment (ztplz) [inline components/form/FormItem.tsx]: if `list` is a empty array, slice(1) will cause a error.
`````
if (list.length) return [];
return list.reduce((current, item) => (
      [...current, ' ', item]
    ), [])
    .slice(1);
``````

**Review by ztplz — APPROVED** [review]

File: components/form/FormItem.tsx (hunk 2 above)
Comment (ztplz) [inline components/form/FormItem.tsx]: `Array<T | string>` `string` is unnecessary.

**Review by ztplz — APPROVED** [review]

File: components/form/FormItem.tsx (hunk 3 above)
Comment (zombieJ) [reply components/form/FormItem.tsx]: Sorry. I miss it. Do you mean `[].slice(1)` cause error?

Comment (ztplz) [reply components/form/FormItem.tsx]: Sorry for my bad. If index out of range, slice will return a empty array. Just test for now.
``````
[].slice(1)
[]
``````

File: components/form/FormItem.tsx (hunk 2 above)
Comment (zombieJ) [reply components/form/FormItem.tsx]: Since we insert string `' '` in the array. Remove `| string` will get error.

**Pull request merged**